File: dataset/human/python/file_1697.py
def main():
    n,m=map(int,input().split(' '))
    a = []
    b = []
    ans = True
    for i in range(n):
        a.append(list(input()))
    for i in range(m):
        b.append(list(input()))
    for i in range(n-m+1):
        for j in range(n-m+1):
            if a[i][j:j+m] == b[0]:
                for k in range(m):
                    if a[i+k][j:j+m] != b[k]:
                        ans = False
                if ans == True:
                    print('Yes')
                    exit()
                ans = True

    print('No')

if __name__ == '__main__':
    main()

File: dataset/human/c/file_1697.c
#include<stdio.h>
int main(void)
{
  int i,ans,b,n;
  int a[20];
  int c[20];
  scanf("%d",&n);
  for(i=0;i<n;i++){
    scanf("%d",&a[i]);
  }
  for(i=0;i<n;i++){
    scanf("%d",&b);
    ans+=b;
  }
  for(i=0;i<n-1;i++){
    scanf("%d",&c[i]);
  }
  for(i=0;i<n-1;i++){
    if (a[i+1]-a[i]==1){
      ans+=c[a[i]-1];
    }
  }
  printf("%d",ans);
  return 0;
}
